File: fixtures/scale/php/large/src/Store/SessionState.php
<?php

namespace CipherLab\Store;

class SessionState
{
    private ResultsStore $store;

    public function __construct()
    {
        $this->store = new ResultsStore();
    }

    public function previousSessions(): int
    {
        $runs = $this->store->priorRuns();
        $families = array_unique(array_map(fn (ResultRecord $r) => $r->family, $runs));
        return count($families) === 0 ? 0 : intdiv(count($runs), count($families));
    }

    public function resultsStore(): ResultsStore
    {
        return $this->store;
    }
}

File: fixtures/scale/swift/large/Store/SessionState.swift
struct SessionState {
    private let store = ResultsStore()

    func previousSessions() -> Int {
        let runs = store.priorRuns()
        let families = Set(runs.map { $0.family })
        return families.isEmpty ? 0 : runs.count / families.count
    }

    func resultsStore() -> ResultsStore {
        return store
    }
}
